### PR TITLE
DEV-11258 Initial commit of the project code

### PR DESCRIPTION
**Pull request by sean-rn**
# Description

Initial commit of the code for the Hellosign limited client SDK

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by EnigmaticMind — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,30 @@
-# go-hellosign
-Client SDK for a subset of the Hellosign (Dropbox Sign) API
+hellosign-sdk
```
Comment (cbaucom) [inline README.md]: nit: maybe rename this to `go-hellosign`? Same thing on line 4.

**Review by cbaucom — APPROVED** [review]

File: client_impl.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
+		respBody, err := io.ReadAll(io.LimitReader(resp.Body, 4096))
+		if err != nil {
+			return fmt.Errorf("error reading error response: %s: %w", resp.Status, err)
+		}
+		return fmt.Errorf("request returned %s: %s", resp.Status, respBody)
+	}
+
+	switch t := target.(type) {
```
Comment (jsalehCRN) [inline client_impl.go:107]: I feel like this is slightly more flexible if you just return the bytes from the request and let each function handle it's response type, but maybe this is more convenient

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module github.com/connectRN/go-hellosign
+
+go 1.21
```
Comment (jsalehCRN) [inline go.mod:3]: is there any reason this can't be 1.23?

**Review by jsalehCRN — APPROVED** [review]

File: client_impl.go (hunk 2 above)
Comment (sean-rn) [reply client_impl.go:107]: That is more flexible, but this is more convenient

File: go.mod (hunk 3 above)
Comment (sean-rn) [reply go.mod:3]: @jsalehCRN Because apiserv is 1.21 and I didn't feel like upgrading apiserv today.

**Pull request merged**